src/main/java/seedu/taassist/logic/commands/AssignCommand.java

```diff
@@ -87,6 +87,6 @@ public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof AssignCommand // instanceof handles nulls
                 && indices.equals(((AssignCommand) other).indices))
-                && moduleClassToAssign.equals(((AssignCommand) other).moduleClassToAssign);
+                && moduleClassToAssign.isSame(((AssignCommand) other).moduleClassToAssign);
```

Comment:
Do we really want this? I think this goes against the philosophy of the `Identity<T>` interface, that `equals` check for exact equality and `isSame` check with identity. In this case, isn't just just like `indices` is the identity of `AssignCommand`?

Discuss: #187.

Reply:
I reverted it back to `equals` in the latest commit. I suggest we merge this PR first and revisit this issue later since this PR also fixes the bug in grade command (I shouldve done this in a separate PR, mb).